yazi-core/src/input/input.rs

```diff
@@ -221,15 +271,30 @@ impl Input {
 		true
 	}
 
+	pub fn forward_delete(&mut self) -> bool {
+		let snap = self.snaps.current_mut();
+		// Return false when there is no character on the right to delete.
+		// Note that the cursor can be at index `snap.value.len()` when in
+		// edit mode.
+		if snap.cursor > snap.value.len() - 1 {
```

Comment:
Pressing `Ctrl-D` when the input is empty will result in a process crash

Reply:
I didn't think about that, thanks for catching!